client/my-sites/themes/theme-showcase.jsx

```diff
@@ -218,7 +197,10 @@ const ThemeShowcase = React.createClass( {
 						label={ this.props.uploadLabel }
 					/>
 				}
-				<ThemesSelection query={ query }
+				<ThemesSelection
+					search={ search }
+					tier={ this.props.tier }
+					vertical={ this.props.vertical }
```

Comment:
I think we're missing the `filter` prop. (Try using a filter in JP single site mode, it doesn't seem to filter the results.)